cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_utils.go

```diff
@@ -143,3 +151,42 @@ func machineSetIsOwnedByMachineDeployment(machineSet *MachineSet, machineDeploym
 	}
 	return false
 }
+
+func scaleFromZeroEnabled(annotations map[string]string) bool {
+	cpu := annotations[cpuKey]
+	mem := annotations[memoryKey]
+	gpu := annotations[gpuKey]
+
+	if cpu != "" && mem != "" && gpu != "" {
+		return true
+	}
+	return false
+}
+
+func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error) {
+	if val, exists := annotations[cpuKey]; exists && val != "" {
+		return resource.ParseQuantity(val)
+	}
+	return zeroQuantity.DeepCopy(), nil
```

Comment:
Wonder if it might be worth extracting this to something like below and calling that from each of the methods listed here since the only difference between them is the key passed into the map to check?
```
func parseKey(annoations map[string]string, key string) (resource.Quantity, error) {
    if val, exists := annotations[key]; exists && val != "" {
	return resource.ParseQuantity(val)
    }
    return zeroQuantity.DeepCopy(), nil
}
```

Reply:
makes good sense, i'll refactor it